src/connection.h

```diff
@@ -245,14 +351,78 @@ int connDisableTcpNoDelay(connection *conn);
 int connKeepAlive(connection *conn, int interval);
 int connSendTimeout(connection *conn, long long ms);
 int connRecvTimeout(connection *conn, long long ms);
-int connPeerToString(connection *conn, char *ip, size_t ip_len, int *port);
-int connFormatFdAddr(connection *conn, char *buf, size_t buf_len, int fd_to_str_type);
-int connSockName(connection *conn, char *ip, size_t ip_len, int *port);
-const char *connGetInfo(connection *conn, char *buf, size_t buf_len);
-
-/* Helpers for tls special considerations */
-sds connTLSGetPeerCert(connection *conn);
-int tlsHasPendingData();
-int tlsProcessPendingData();
+
+typedef enum {
+    /* TLS: set configure. @arg0: pointer(void *) to config of redisTLSContextConfig; @arg1: reconfigure(bool) to overwrite previous configure. also see tlsConfigure */
+    CTRL_TLS_SET_CONFIG,
+    /* TLS: fetch the peer certificate. @arg0: pointer(connection *) to connection; @arg1: pointer(sds *) to result. see connTLSGetPeerCert */
+    CTRL_TLS_GET_PEER_CERT,
+    /* TLS: fetch ctx. @arg0: pointer(SSL_CTX **) to result. */
+    CTRL_TLS_GET_CTX,
+    /* TLS: fetch client ctx. @arg0: pointer(SSL_CTX **) to result. */
+    CTRL_TLS_GET_CLIENT_CTX,
+} connControlType;
+
+int connControl(ConnectionType *ct, unsigned long cmd, unsigned long arg0, unsigned long arg1, unsigned long arg2, unsigned arg3);
+
+/* Initialize the redis connection framework */
+int connTypeInitialize();
+
+/* Register a connection type into redis connection framework */
+int connTypeRegister(ConnectionType *ct);
+
+/* Lookup a connection type by type name */
+ConnectionType *connectionByType(const char *typename);
+
+/* Fast path to get TCP connection type */
+ConnectionType *connectionTypeTcp();
+
+/* Fast path to get TLS connection type */
+ConnectionType *connectionTypeTls();
+
+/* Fast path to get Unix connection type */
+ConnectionType *connectionTypeUnix();
+
+/* Lookup the index of a connection type by type name, return -1 if not found */
+int connectionIndexByType(const char *typename);
+
+/* Create a connection of specified type */
+static inline connection *connCreate(ConnectionType *ct) {
+    return ct->conn_create();
+}
+
+/* Create a accepted connection of specified type.
+ * @priv is connection type specified argument */
```

Comment:
I don't think @ is used anywhere in Redis for describing arugments. 
```suggestion
 * priv is connection type specified argument */
```

There is a couple of other places this is set as well, I didn't call them all out though.